File: BackEnd/ApiCriminalidade/ApiCriminalidade.Domain/Entities/Assalto.cs
using ApiCriminalidade.Domain.Entities;

namespace ApiCriminalidade.Domain.Entities
{
    // sealed serve para dizer que está classe não pode ser herdada
    public sealed class Assalto : Entity
    {

        public int Id { get; set; }

        public int QuantidadeAgressores { get; set; }

        public bool PossuiArma { get; set; }

        public int OcorrenciaId { get; set; }

        public Ocorrencia? Ocorrencia { get; set; }

        public int TipoArmaId { get; set; }

        public TipoArma? TipoArma { get; set; }

        public IEnumerable<AssaltoTipoBem> AssaltosTipoBens { get; set; }

        
    }
}

File: FrontEnd/lib/Models/Assalto.dart
class Assalto{

  Assalto(
      this._tipoBens,
      this._descricao,
      this._dataHora,
      this._cadastrouBoletimOcorrencia,
      this._quantidadeAgressores,
      this._possuiArma,
      this._tipoArma);

  List<int> _tipoBens;

  String _descricao;

  String _dataHora;

  bool _cadastrouBoletimOcorrencia;

  int _quantidadeAgressores;


  bool _possuiArma;

  int _tipoArma;


  int get quantidadeAgressores => _quantidadeAgressores;

  set quantidadeAgressores(int value) {
    _quantidadeAgressores = value;
  }

  bool get cadastrouBoletimOcorrencia => _cadastrouBoletimOcorrencia;

  set cadastrouBoletimOcorrencia(bool value) {
    _cadastrouBoletimOcorrencia = value;
  }

  String get dataHora => _dataHora;

  set dataHora(String value) {
    _dataHora = value;
  }

  String get descricao => _descricao;

  set descricao(String value) {
    _descricao = value;
  }

  List<int> get tipoBens => _tipoBens;

  set tipoBens(List<int> value) {
    _tipoBens = value;
  }

  bool get possuiArma => _possuiArma;

  set possuiArma(bool value) {
    _possuiArma = value;
  }

  int get tipoArma => _tipoArma;

  set tipoArma(int value) {
    _tipoArma = value;
  }
}